theano/misc/pycuda_example.py

```diff
@@ -285,6 +285,11 @@ def __init__(self, scalar_op, inplace_pattern=None, name=None):
         self.scalar_op = scalar_op
         self.inplace_pattern = inplace_pattern
 
+    # As we have a dict in props, we need to implement __hash__
+    def __hash__(self):
+        return (hash(type(self)) ^ hash(self.scalar_op) ^
+                hash_from_dict(self.inplace_pattern))
```

Comment:
I tried to make the parent \__hash__ work, but I wasn't able to make it work. So I fall back to the old code that was working.

I updated the PR to make one hash of a tuple to remove the xor.